WIP: Clean setup

davidungar opened the pull request:
<!-- What's in this pull request? -->
Reorganize CompilerInstance::setup() on the way to multiple primaries.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->


<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

davidungar committed:
Factor code from tools & move input conversion.

Fix StringRef parameter declaration.

Read command line arguments even if filelist is present.

Don’t clearPrimaryInputs for swift-ide-test, set primary to first file.

Keep separate lists instead of same file twice in one list.

Address Jordan’s comments:

- Also be consistent about has vs have for Input predicates

Fix bug.

davidungar committed:
Bring over redone output files and broken up ParseFrontendOptions from older PR.

davidungar commented:
@swift-ci please smoke test OS X platform

davidungar commented:
@swift-ci please smoke test OS X platform

davidungar commented:
@swift-ci please smoke test os x platform

compnerd commented on tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp:
```diff
@@ -761,27 +754,27 @@ ASTUnitRef ASTProducer::createASTUnit(SwiftASTManager::Implementation &MgrImpl,
+                                                             Content.Buffer = llvm::MemoryBuffer::getNewMemBuffer(0, File);
+                                                           }
+                                                           Contents.push_back(std::move(Content));
+                                                         });
```
Just a drive-by.  This is pretty far indented.  Would be nice to just hoist this out and create a local variable for the lambda which you can pass as an argument.  It would significantly un-indent the code here.  There is at least one other instance of this.

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar committed:
Redo Inputs data structure.

Formatted.

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar commented:
@swift-ci please smoke test os x platform

davidungar committed:
Fix cosmetics.

davidungar committed:
Fix duplicate input file in test.

davidungar committed:
Fix bug I introduced in SwiftASTManager.cpp.

davidungar committed:
Changes from review of previous pull-request.

To the extent still applicable:
Change “has” routines to “have.”
Use more consistent casing.
Remove spurious “DelayedFunctionParsing” option.
Move debugFail routines to top lexical level.
Rename and reorder declaration of functions in FrontendArgsToOptionsConverter.
Move, reword, and doxygenate comments for some of those functions.
Fix casing on some more setUp* functions.
Return NoneAction instead of existing RequestedAction in FrontendArgsToOptionsConverter::determineRequestedAction.
Remove test names and put in FIXME’s.
Remove “Jordan” from comments & reword.
Reorder if-then arms of FrontendArgsToOptionsConverter::computeOutputFilenames for readability.
Test for empty string instead of equality with “”.
Use hasUnusedModuleDocOutputPath.
Remove optionality from return type of getOutputFilenamesFromCommandLineOrFilelist.
Rename isPrimaryInputAFileAt to isThereAPrimaryInputWithAFilenameAt.
Added a FIXME in doesActionProduceOutput to reflect that some actions actually do not produce output.
